client/state/data-layer/wpcom/sites/posts/likes/mine/delete/index.js

```diff
@@ -0,0 +1,48 @@
+/** @format */
+
+/**
+ * External Dependencies
+ */
+
+/**
+ * Internal Dependencies
+ */
+import { like, updateLikeCount } from 'state/posts/likes/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { POST_UNLIKE } from 'state/action-types';
+import { bypassDataLayer } from 'state/data-layer/utils';
+
+export function fromApi( response ) {
+	return {
+		success: !! response.success,
```

Comment:
instead of passing through a success, i think this should throw in the case of failure.

```js
export function fromApi( response ) {
  if ( ! response || ! response.success ) {
    throw new Error('unsuccesful post unlike');
  }
}
```

Reply:
done. rejiggered it quite a bit too.